### PR TITLE
Update run-a-terraform-plan-with-the-terraform-apply-step.md

### PR DESCRIPTION
**Pull request by thisrohangupta**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by deepakbolangady — DISMISSED** [review]
Approving after making several changes:
* Edits
* The new section needed to be fleshed out in the form of numbered steps. Did that.
* The new section needed to be extracted and placed in a new file, and then referenced from the documentation for the Plan, Apply, and Destroy steps.
* The Destroy step's documentation did not have the TF var content, so added a new reference to the extracted content. 

@vlica-harness and @thisrohangupta, please review these changes and let me know if it's good to merge.

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/optional-tf-var-files.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                            spec:
+                              optional: true
+                              store:
+                                type: Github
+                                spec:
+                                  gitFetchType: Branch
+                                  repoName: ""
+                                  branch: main
+                                  paths:
+                                    - terraform/localresource/localresourcevars/localresource.tfvar
```
Comment (vlica-harness) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/optional-tf-var-files.md:117]: can we rename the file extension here from tfvar to tfvars  terraform/localresource/localresourcevars/localresource.tfvar

File: docs/continuous-delivery/cd-infrastructure/terraform-infra/optional-tf-var-files.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+The ability to mark Terraform variable files as optional files is behind the feature flag `CDS_TERRAFORM_SUPPORT_OPTIONAL_VAR_FILE_PATHS_NG`. To enable the feature, contact [Harness Support](mailto:support@harness.io).
+:::
+
+Harness supports all [Terraform input types and values](https://www.terraform.io/docs/language/expressions/types.html).
+
+## Harness Manager
+To specify optional Terraform variable files, do the following:
+
+:::tip 
+In some fields in this workflow, you can specify a fixed value or expression or set up the field to accept an input at run time. For information about value types and how to identify fields that offer this functionality, go to [Fixed values, runtime inputs, and expressions](/docs/platform/variables-and-expressions/runtime-inputs).
```
Comment (vlica-harness) [inline docs/continuous-delivery/cd-infrastructure/terraform-infra/optional-tf-var-files.md:22]: "workflow" to be renamed with "pipeline" maybe?

Comment (deepakbolangady) [reply docs/continuous-delivery/cd-infrastructure/terraform-infra/optional-tf-var-files.md:22]: This new tip I created is an attempt to standardize instructions for the value type functionality across the doc set. Several fields have this functionality, and it's not restricted to pipelines only. If "workflow" conflicts with another idea, I can replace this word with "procedure". In fact, "procedure" is a better word, so thank you for pointing this out.

**Pull request merged**